launch/launch/launch_description.py

```diff
@@ -45,12 +48,23 @@ class LaunchDescription(LaunchDescriptionEntity):
     def __init__(
         self,
         initial_entities: Optional[Iterable[LaunchDescriptionEntity]] = None,
+        *,
+        deprecated_reason: Optional[Text] = None
     ) -> None:
         """Constructor."""
         self.__entities = list(initial_entities) if initial_entities is not None else []
+        self.__deprecated_reason = deprecated_reason
 
     def visit(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
         """Override visit from LaunchDescriptionEntity to visit contained entities."""
+        if 'current_launch_file_path' in context.get_locals_as_dict():
```

Comment:
What about the else case here? Maybe just a runtime error? I think it should never happen.

Reply:
I think it happens if you create a `LaunchDescription` in a script and the you manually create a  `LaunchService` with it, since `current_launch_file_path` local is added by `IncludeLaunchDescription` action (if you don't use `IncludeLaunchDescription`, that local doesn't exist).

Maybe, logging an std warning without the file path would be a good idea.
I will do that.